components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -146,42 +138,39 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
         this.isMinorV1 = version === 'v1';
       });
 
-    this.store.select(assignableProjects)
-      .subscribe((assignable: ProjectsFilterOption[]) => {
-        assignable.forEach(({ value: id, label: name, type: stringType }) => {
-          const type = <IAMType>stringType;
-          // TODO (tc) Don't actually have the status queried in here but
-          // also don't need it for this page. Maybe the status field
-          // should be optional in the project model?
-          const proj: Project = { id, name, type, status: 'NO_RULES' };
-          // we don't want to override projects that we fetched
-          // that were part of the team already
-          if (!this.projects[proj.id]) {
-            const checked = false;
-            this.projects[proj.id] = { ...proj, checked };
-          }
-        });
-      });
-
-    this.store.select(iamMajorVersion)
-      .pipe(
-        takeUntil(this.isDestroyed),
-        filter(identity)
-        )
-      .subscribe((version) => {
-        this.isMajorV1 = version === 'v1';
-
-        // Triggered every time the team is updated.
-        if (this.isMajorV1) {
-          this.store.select(v1TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getTeamDependentData.bind(this));
-        } else {
-          this.store.select(v2TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getTeamDependentData.bind(this));
-        }
-      });
+    combineLatest([
+      this.store.select(assignableProjects),
+      this.store.select(optionsLoadingStatus),
+      this.store.select(v1TeamFromRoute),
+      this.store.select(v2TeamFromRoute),
+      this.store.select(iamMajorVersion)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      map(([assignable, pStatus, v1Team, v2Team, major]:
+        [ProjectsFilterOption[], EntityStatus, Team, Team, IAMMajorVersion]) => {
+        this.isMajorV1 = major === 'v1';
+        return [assignable, pStatus, (this.isMajorV1 ? v1Team : v2Team)];
+      }),
+      filter(([_assignable, pStatus, team]:
+        [ProjectsFilterOption[], EntityStatus, Team]) =>
+        pStatus !== EntityStatus.loading && !!team
+      ),
+      map(([assignable, _pStatus, team]) => {
+        this.team = team;
+
+        assignable
+          .forEach(({ value: id, label: name, type: stringType }) => {
+            const type = <IAMType>stringType;
```

Comment:
because that's not the type of `stringType`, it's string 😉 We'll have to do some kind of casting, or, maybe better yet, create a type guard for the string => IAMType conversion 🤔